### ISSUE TITLE
Inconsistent types detected 

### ISSUE THREAD

**Issue by amuino** [apiaryio/drafter#245]
We are getting the following error running Drafter:
```
$ drafter --validate --use-line-num sample.apib.md

OK.
warning: (4)  ExtendElement must be composed from Elements of same type
```

This is the `sample.apib.md`
```
FORMAT: 1A

# My Public API
## Request [GET /some/endpoint]
+ Response 200 (application/json; charset=utf-8)
  + Attributes(Booking State Response)

# Data Structures
## Point (array)
## Booking State Params
## Booking State Response
- loc (Point)
## O1
## O2
## O3
## O4
## O5
## O6
## O7
## O8
## O9
## O10
## O11
## O12
## O13
## O14
```

After adding some debug output and rebuilding, the issue arises because when reaching [`ExtendElement::push_back`](https://github.com/apiaryio/drafter/blob/master/src/refract/Element.cc#L167-L188) we have `baseType` identified as an `ArrayElement` but `type` is an `ObjectElement`.

This seems to be caused by some previous error that I could not trace back (I'm not proficient at C++).

Just by removing `O14` from the blueprint, the `type` is correctly detected as `ArrayElement` and no errors are detected. They are also solved by some renaming of the `Booking State Response` structure, for instance shortening it to `Response`.

This is related to https://github.com/apiaryio/dredd/issues/381 and https://github.com/danielgtaylor/aglio/issues/237

**Comment by zdne** [comment]
Duplicate see #217

**Comment by kylef** [comment]
@amuino This was fixed in [Drafter 2.3.0-pre.0](https://github.com/apiaryio/drafter/releases/tag/v2.3.0-pre.0), would you be able to give that a go and confirm you're no longer seeing this.

**Comment by amuino** [comment]
@kylef thanks for the heads up. How can I test it? I can't see the version [on npm](https://www.npmjs.com/package/drafter)

**Comment by amuino** [comment]
@kylef verified that it is solved in 2.3.0-pre.0. Thanks for the heads up!
